bouncerscript/task.py

```diff
@@ -55,9 +54,8 @@ def get_supported_actions(script_config):
     return tuple(script_config['schema_files'].keys())
 
 
-def validate_task_schema(context, action):
+def validate_task_schema(context):
     """Perform a schema validation check against taks definition"""
-    schema_file = context.config['schema_files'][action]
-    with open(schema_file) as fh:
-        task_schema = json.load(fh)
-    scriptworker.client.validate_json_schema(context.task, task_schema)
+    action = get_task_action(context.task, context.config)
```

Comment:
Nit: `context.action` is set [2 lines after](https://github.com/mozilla-releng/bouncerscript/pull/15/files#diff-835b68f16b034323ff73ec1722407044R64) the call to `validate_task_schema()`. I think we should use this one.

Reply:
Not sure what you mean here, could you please elaborate?

Before this PR, the schema validation was done after we set the server/action within the context. But I glanced on your previous PRs' in signingscript/beetmoverscript/shipitscript and saw that you always did the validation beforehand, which makes more sense to me indeed. As you don't want to proceed with any other logic absent a correct task definition. Hence, I've changed the order. But since the `context.action` is set afterwards, I couldn't find another way but to use the call again. I guess the logic is saner in this case but has the drawback of calling `get_action_task` twice. 

There's a similar approach in the beetmoverscript PR, [here](https://github.com/mozilla-releng/beetmoverscript/pull/108/files#diff-21f3f939e3375901322bd3639f4394bcL24) and [here ](https://github.com/mozilla-releng/beetmoverscript/pull/108/files#diff-98ba072d44cc67fadf38dc459a83d934R218)

Reply:
Oh, I hadn't noticed context.action was set in beetmoverscript too. Thanks for pointing it out!

I don't have a strong opinion on this. I agree with you, safer is better. Calling this function twice is not a big deal.

Reply:
Blame my copy-paste skills for this :) I'm like "Hm, how did Johan solve this problem? Ah, I like that better, let me import that knowledge!" 😄